scenes/game_state/game_state.gd

```diff
@@ -3,5 +3,42 @@
 @tool
 extends Node
 
+## Emited when a new item is collected, even if it wasn't added to the
+## inventory due to it being already there.
+signal item_collected(item: InventoryItem)
+
+## Global inventory, used to track the items the player obtains and that
+## can be added to the loom.
 @export var inventory: Inventory = Inventory.new()
 @export var current_spawn_point: NodePath
+
+## Quest's items. Used to track the progress withing a story quest.
+var story_quest_inventory: Inventory = Inventory.new()
+
+
+## Resets the [member story_quest_inventory]. This needs to be called when
+## a new story quest starts.
+func quest_started() -> void:
```

Comment:
The function name looks like is going to return a boolean. What about `start_quest()` or similar?